### PR TITLE
Update capitalization of "Mbed" and fix bump_version.sh

### PR DESCRIPTION
**Pull request by gilles-peskine-arm**
* Update the spelling of "Mbed TLS". The official spelling used to be "mbed TLS" but changed a few years ago and we haven't caught up. Keep the old spelling in places where it can change things: where it gets embedded in test certificates, and in an imported file.
* Fix a bug in `bump_version.sh` (already noticed in the 3.4.0 release) that caused it to miss a file due to the spelling change.

## PR checklist

Please tick as appropriate and edit the reasons (e.g.: "backport: not needed because this is a new feature")

- [x] **changelog** not required
- [x] **backport** https://github.com/Mbed-TLS/mbedtls/pull/8194
- [x] **tests** not required

### CODE REVIEW

**Review by tgonzalezorlandoarm — DISMISSED** [review]

**Comment by daverodgman** [comment]
This misses `programs/x509/cert_write` and `programs/x509/cert_req` - is that intentional?

**Comment by gilles-peskine-arm** [comment]
@daverodgman [yes](https://github.com/Mbed-TLS/mbedtls/pull/8034/commits/dad6f5adcb9ab0c2df45464179a5cd776c9db290#prev-commit-dad6f5adcb9ab0c2df45464179a5cd776c9db290)

**Review by daverodgman — DISMISSED** [review]

File: include/mbedtls/error.h [hunk 1]
```diff
@@ -166,7 +166,7 @@ static inline int mbedtls_error_add(int high, int low,
 }
 
 /**
- * \brief Translate a mbed TLS error code into a string representation,
+ * \brief Translate a Mbed TLS error code into a string representation,
```
Comment (tom-cosgrove-arm) [inline include/mbedtls/error.h:169]: (pre-existing, but might as well fix while here) `an Mbed TLS error code`

**Comment by tom-cosgrove-arm** [comment]
Should we not also have

```
diff --git a/tests/data_files/Makefile b/tests/data_files/Makefile
index eff44d8ac..fc23e3f28 100644
--- a/tests/data_files/Makefile
+++ b/tests/data_files/Makefile
@@ -1,7 +1,7 @@
 ## This file contains a record of how some of the test data was
 ## generated. The final build products are committed to the repository
 ## as well to make sure that the test data is identical. You do not
-## need to use this makefile unless you're extending mbed TLS's tests.
+## need to use this makefile unless you're extending Mbed TLS's tests.

 ## Many data files were generated prior to the existence of this
 ## makefile, so the method of their generation was not recorded.
```

**Comment by tom-cosgrove-arm** [comment]
Note: conflicts

**Comment by gilles-peskine-arm** [comment]
There were merge conflicts so I rebased. I applied the same script from the changelog message to re-do the big capitalization commit.

**Review by daverodgman — APPROVED** [review]

**Review by tom-cosgrove-arm — APPROVED** [review]
LGTM

**Pull request merged**